tutorials/SingleQubitGates/SingleQubitGates.ipynb

```diff
@@ -0,0 +1,874 @@
+    "\n",
+    "$$X = |0\\rangle\\langle1| + |1\\rangle\\langle0| \\\\\n",
+    "|0\\rangle\\langle1| + |1\\rangle\\langle0| =\n",
+    "\\begin{bmatrix} 1 \\\\ 0 \\end{bmatrix}\\begin{bmatrix} 0 & 1 \\end{bmatrix} +\n",
+    "\\begin{bmatrix} 0 \\\\ 1 \\end{bmatrix}\\begin{bmatrix} 1 & 0 \\end{bmatrix} =\n",
+    "\\begin{bmatrix} 0 & 1 \\\\ 0 & 0 \\end{bmatrix} + \\begin{bmatrix} 0 & 0 \\\\ 1 & 0 \\end{bmatrix} =\n",
+    "\\begin{bmatrix} 0 & 1 \\\\ 1 & 0 \\end{bmatrix}$$\n",
+    "\n",
+    "This can also be used for calculations:\n",
+    "\n",
+    "$$X|0\\rangle = \\big(|0\\rangle\\langle1| + |1\\rangle\\langle0|\\big)|0\\rangle = |0\\rangle\\langle1|0\\rangle + |1\\rangle\\langle0|0\\rangle = |0\\rangle\\big(\\langle1|0\\rangle\\big) + |1\\rangle\\big(\\langle0|0\\rangle\\big) = |0\\rangle(0) + |1\\rangle(1) = |1\\rangle$$\n",
+    "\n",
+    "> That last step may seem a bit confusing. Recall that $|0\\rangle$ and $|1\\rangle$ form an **orthonormal basis**. That is, they are both normalized, and they are orthogonal to each other.\n",
+    ">\n",
+    "> A vector is normalized if its norm is equal to $1$, which only happens if its inner product with itself is equal to $1$. This means that $\\langle0|0\\rangle = \\langle1|1\\rangle = 1$\n",
+    ">\n",
+    "> Two vectors are orthogonal to each other if their inner product equals $0$. This means that $\\langle0|1\\rangle = \\langle 1|0\\rangle = 0$."
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "> ## Ket-bra decomposition\n",
+    ">\n",
+    "> This section deals with finding the ket-bra decompositions of quantum gates. This section is not necessary to start working with quantum gates, so feel free to skip it for now, and come back to it later.\n",
+    ">\n",
+    "> You can use the properties of [eigenvalues and eigenvectors](../LinearAlgebra/LinearAlgebra.ipynb#Eigenvalues-and-Eigenvectors) to find the ket-bra decomposition of any gate. Given a gate $A$, and its eigenvectors $|\\phi\\rangle$ and $|\\psi\\rangle$, if:\n",
+    ">\n",
+    "> $$A|\\phi\\rangle = \\epsilon|\\phi\\rangle \\\\\n",
+    "A|\\psi\\rangle = \\zeta|\\psi\\rangle$$\n",
+    ">\n",
+    "> Then:\n",
+    ">\n",
+    "> $$A = \\epsilon|\\phi\\rangle\\langle\\phi| + \\zeta|\\psi\\rangle\\langle\\psi|$$\n",
+    ">\n",
+    "> Let's use our $X$ gate as a simple example. The $X$ gate has two eigenvectors: $|+\\rangle = \\frac{1}{\\sqrt{2}}\\big(|0\\rangle + |1\\rangle\\big)$ and $|-\\rangle = \\frac{1}{\\sqrt{2}}\\big(|0\\rangle - |1\\rangle\\big)$. Their eigenvalues are $1$ and $-1$ respectively:\n",
+    ">\n",
+    "> $$X|+\\rangle = |+\\rangle \\\\\n",
+    "X|-\\rangle = -|-\\rangle$$\n",
+    ">\n",
+    "> Here's what the decomposition looks like:\n",
+    ">\n",
+    "> $$X = |+\\rangle\\langle+| - |-\\rangle\\langle-| =\n",
```

Comment:
I think this calculation can be made a bit smaller without losing clarity. Multiline suggestions are not great, but this should work:

    > $$X = |+\rangle\langle+| - |-\rangle\langle-| = \\
    = \frac{1}{2}\big[\big(|0\rangle + |1\rangle\big)\big(\langle0| + \langle1|\big) - \big(|0\rangle - |1\rangle\big)\big(\langle0| - \langle1|\big)\big] = \\
    = \frac{1}{2}\big(\color{red}{|0\rangle\langle0|} + |0\rangle\langle1| + |1\rangle\langle0| + \color{red}{|1\rangle\langle1|} - \color{red}{|0\rangle\langle0|} + |0\rangle\langle1| + |1\rangle\langle0| - \color{red}{|1\rangle\langle1|}\big) = \\
    = \frac{1}{2}\big(2|0\rangle\langle1| + 2|1\rangle\langle0|\big) = |0\rangle\langle1| + |1\rangle\langle0|$$

for the following image:

![image](https://user-images.githubusercontent.com/10113024/67260647-49d90b00-f451-11e9-88d6-d1a94daa95bb.png)

Using `\cancel` command would be even better, but I'm not sure notebooks support opening packages...